addons/mail/static/src/components/message/message.xml

```diff
@@ -77,7 +77,9 @@
                             </t>
                             <t t-if="message.date">
                                 <div class="o_Message_date o_Message_headerDate" t-att-class="{ 'o-message-selected': isSelected }" t-att-title="datetime">
-                                    - <t t-esc="message.dateFromNow"/>
+                                    <span t-if="message.author_has_user and message.sender and message.author and message.sender.id != message.author.id">
```

Comment:
make comparison on records (`sender` vs `author`) rather than record fields (`sender.id` vs `author.id`)

Reply:
Also should use `!==` operator in JS.